src/components/page.tsx

```diff
@@ -4,9 +4,17 @@ export function PageHeader({children}: {children: ReactNode}) {
   return <header className="mb-5">{children}</header>;
 }
 
-export function PageTitle({children}: {children: ReactNode}) {
+interface PageTitleProps {
+  children: ReactNode;
+  'data-test'?: string;
```

Comment:
Curious, because I don't have a lot of experience with Cypress: is it common to add properties to interfaces and data structures just for testing? Even though it's an optional element, a user of the component may not expect it or find it difficult to understand (e.g. what's it for)?

Reply:
I've added a comment, to make it clearer.

Reply:
Ok, I think I now understand the question, but I don't really know the answer and I can't really find good resources. So this is my proposal.

We can try to use dynamic test ids for components as little as possible. For example, most pages will properly only have one PageTitle so we could have a fixed data-test for this component, like "pageTitle". If there is more than one component you can also first use a selector in the wrapper. (cy.getBySel('sectionName').getBySel('pageTitle'))

If we do need a dynamic test id (for the filters, I will need them). I do like using `data-test` as a component attribute. In other projects, I have used other attribute names like `testId` for components. I personally did not like having a different name for components than the HTML tags.

But then we have a problem if we want to pass more than one attribute selector. I don't think this is needed often. Because you can have one dynamic test id at the top level and multiple fixed ones at lower levels. I think in most cases that will be enough. But if that is not enough, only in those cases we can use attributes like `testId`, and `testIdDetails`, so we can add more than one.

Reply:
Thank you - a clear explanation. Your proposal makes sense!